debian/cargo-checksum.json

```diff
@@ -1 +1 @@
-{"package":"Could not get crate checksum","files":{}}
+{"package":"2955b1fe31e1fa2fbd1976b71cc69a606d7d4da16f6de3333d0c92d51419aeff","files":{}}
```

Comment:
How do you compute the checksum? That's something I was unable to find!

Reply:
> How do you compute the checksum? That's something I was unable to find!

That's easy. You don't actually need to compute anything:

https://github.com/ubuntu/adsys/blob/dc51c88f909be948ce30379c9ed282850d1201d7/Cargo.lock#L373-L377

Reply:
You might want to document this, then, as it's a good way to have a hard-to-understand FTBFS in the future. Maybe a comment in d/rules ?

Reply:
> You might want to document this, then, as it's a good way to have a hard-to-understand FTBFS in the future. Maybe a comment in d/rules ?

Okay, I will add a comment in `d/rules` to explain this.

Reply:
Even though I like seeing the rules file clean again, the scalability of this suggestion worries me. Right now, this is a problem related only to the system-deps dependency, but this can change. I'm definitely not an expert, but I'm assuming this checksum could change if the version changes and then we would need to update this yet again.
Yes, it's ugly. But tricking cargo into thinking there is no lockfile at build time still seems to be the better solution to allow build and still scale.

Reply:
Done. Comment addressed: added new comments to explain some of the more obscure workarounds.

Reply:
> Even though I like seeing the rules file clean again, the scalability of this suggestion worries me. Right now, this is a problem related only to the system-deps dependency, but this can change. I'm definitely not an expert, but I'm assuming this checksum could change if the version changes and then we would need to update this yet again.
> Yes, it's ugly. But tricking cargo into thinking there is no lockfile at build time still seems to be the better solution to allow build and still scale.

We can see how this goes in the future: if this becomes unmanageable, we can restore the original behaviour.

After going through all the dependencies of this package, I think the only issue will probably be `gio` and `glib` (because `system-deps` is a transitive dependency of them), and they don't add dependencies that often. I believe this is going to be fine for quite a while.

Reply:
Yes, I think that with the action this should be fine, specially since I don't think there will be more Rust on adsys anytime soon and until there we can probably hope that cargo provides us with a bit more overridable options for this stuff.
Thanks for taking some time with this!